src/main/java/org/opensearch/knn/index/VectorDataType.java

```diff
@@ -0,0 +1,192 @@
+                VECTOR_DATA_TYPE,
+                vectorDataType,
+                supportedTypes
+            )
+        );
+    }
+
+    /**
+     * Validate the float vector values if it is a number and in the finite range.
+     *
+     * @param value  float vector value
+     */
+    public static void validateFloatVectorValues(float value) {
+        if (Float.isNaN(value)) {
+            throw new IllegalArgumentException("KNN vector values cannot be NaN");
+        }
+
+        if (Float.isInfinite(value)) {
+            throw new IllegalArgumentException("KNN vector values cannot be infinity");
+        }
+    }
+
+    /**
+     * Validate the float vector value in the byte range if it is a finite number,
+     * with no decimal values and in the byte range of [-128 to 127].
+     *
+     * @param value  float value in byte range
+     */
+    public static void validateByteVectorValues(float value) {
+        validateFloatVectorValues(value);
+        if (value % 1 != 0) {
+            throw new IllegalArgumentException(
+                "[data_type] field was set as [byte] in index mapping. But, KNN vector values are floats instead of byte integers"
+            );
+        }
+        if ((int) value < Byte.MIN_VALUE || (int) value > Byte.MAX_VALUE) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "[%s] field was set as [%s] in index mapping. But, KNN vector values are not within in the byte range [{}, {}]",
+                    VECTOR_DATA_TYPE,
+                    VectorDataType.BYTE.getValue(),
+                    Byte.MIN_VALUE,
+                    Byte.MAX_VALUE
+                )
+            );
+        }
+    }
+
+    /**
+     * Validate if the given vector size matches with the dimension provided in mapping.
+     *
+     * @param dimension dimension of vector
+     * @param vectorSize size of the vector
+     */
+    public static void validateVectorDimension(int dimension, int vectorSize) {
+        if (dimension != vectorSize) {
+            String errorMessage = String.format("Vector dimension mismatch. Expected: %d, Given: %d", dimension, vectorSize);
+            throw new IllegalArgumentException(errorMessage);
+        }
+
+    }
+
+    /**
+     * Validates and throws exception if data_type field is set in the index mapping
+     * using any VectorDataType (other than float, which is default) with any engine (except lucene).
+     *
+     * @param knnMethodContext KNNMethodContext Parameter
+     * @param vectorDataType VectorDataType Parameter
+     */
+    public static void validateVectorDataType_Engine(
+        ParametrizedFieldMapper.Parameter<KNNMethodContext> knnMethodContext,
+        ParametrizedFieldMapper.Parameter<VectorDataType> vectorDataType
+    ) {
+        if (vectorDataType.getValue() != DEFAULT_VECTOR_DATA_TYPE
+            && (knnMethodContext.get() == null || knnMethodContext.getValue().getKnnEngine() != KNNEngine.LUCENE)) {
```

Comment:
1. There is no difference between both.
2. If knnMethodContext is null, then it will use the default KnnEngine which is NMSLIB. 

But, I have updated the condition to compare it with default engine as it might get changed in the future.